source/particle/property/cpo_elastic_tensor.cc

```diff
@@ -0,0 +1,273 @@
+                                                                      data);
+
+        Particle::Property::CpoElasticTensor<dim>::set_elastic_tensor(data_position,
+                                                                      data,
+                                                                      C_average);
+
+
+      }
+
+
+
+      template <int dim>
+      SymmetricTensor<2,6>
+      CpoElasticTensor<dim>::get_elastic_tensor(unsigned int cpo_data_position,
+                                                const ArrayView<double> &data)
+      {
+        SymmetricTensor<2,6> elastic_tensor;
+        for (unsigned int i = 0; i < SymmetricTensor<2,6>::n_independent_components ; ++i)
+          elastic_tensor[SymmetricTensor<2,6>::unrolled_to_component_indices(i)] = data[cpo_data_position + i];
+        return elastic_tensor;
+      }
+
+
+
+      template <int dim>
+      void
+      CpoElasticTensor<dim>::set_elastic_tensor(unsigned int cpo_data_position,
+                                                const ArrayView<double> &data,
+                                                SymmetricTensor<2,6> &elastic_tensor)
+      {
+        for (unsigned int i = 0; i < SymmetricTensor<2,6>::n_independent_components ; ++i)
+          data[cpo_data_position + i] = elastic_tensor[SymmetricTensor<2,6>::unrolled_to_component_indices(i)];
+      }
+
+
+
+      template <int dim>
+      UpdateTimeFlags
+      CpoElasticTensor<dim>::need_update() const
+      {
+        return update_output_step;
+      }
+
+      template <int dim>
+      UpdateFlags
+      CpoElasticTensor<dim>::get_needed_update_flags () const
+      {
+        return update_default;
+      }
+
+      template <int dim>
+      std::vector<std::pair<std::string, unsigned int>>
+      CpoElasticTensor<dim>::get_property_information() const
+      {
+        std::vector<std::pair<std::string,unsigned int>> property_information;
+
+        property_information.push_back(std::make_pair("cpo_elastic_tensor",SymmetricTensor<2,6>::n_independent_components));
+
+        return property_information;
+      }
+
+      template <int dim>
+      void
+      CpoElasticTensor<dim>::declare_parameters (ParameterHandler &)
+      {}
+
+
+      template <int dim>
+      void
+      CpoElasticTensor<dim>::parse_parameters (ParameterHandler &prm)
+      {
+
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Particles");
+          {
+            prm.enter_subsection("Crystal Preferred Orientation");
+            {
+              n_grains = prm.get_integer("Number of grains per particle");
+              prm.enter_subsection("Initial grains");
+              {
+                n_minerals = dealii::Utilities::split_string_list(prm.get("Minerals")).size();
+              }
+              prm.leave_subsection();
+            }
+            prm.leave_subsection();
+          }
+          prm.leave_subsection ();
+        }
+        prm.leave_subsection ();
+
```

Comment:
Remove one or both of these empty lines, it looks unusual to have two empy lines at the end of a function but before the closing brace.